### PR TITLE
packetbeat application layer analyzers -> pluggable modules

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

**Comment by monicasarbu** [comment]
Great work! LGTM.

File: packetbeat/main.go [hunk 1]
```diff
@@ -5,6 +5,17 @@ import (
 
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/packetbeat/beater"
+
+	// import support protocol modules
+	_ "github.com/elastic/beats/packetbeat/protos/amqp"
```
Comment (ruflin) [inline packetbeat/main.go]: We should automate this in the future. Also on my list for metricbeat, so perhaps we will be able to use the same script.

File: packetbeat/protos/http/http.go [hunk 2]
```diff
@@ -76,29 +77,51 @@ var (
...
+
+func (http *HTTP) setFromConfig(config *httpConfig) {
+	http.Ports = config.Ports
+	http.SendRequest = config.SendRequest
+	http.SendResponse = config.SendResponse
 	http.HideKeywords = config.Hide_keywords
-	if config.Redact_authorization != nil {
-		http.RedactAuthorization = *config.Redact_authorization
-	}
+	http.RedactAuthorization = config.Redact_authorization
```
Comment (ruflin) [inline packetbeat/protos/http/http.go]: One day we will rename this variables ;-)

**Comment by ruflin** [comment]
This change feels really good. It is quite large and which makes it hard to follow as lots of things moved around. I focused more on if the tests stayed the same as there were much fewer changes on the test side. It LGTM and I suggest to merge this one as soon as the builds go green.

File: packetbeat/main.go (hunk 1 above)
Comment (urso) [inline packetbeat/main.go]: not sure I'd want to full automate this. nice thing about having imports in main.go only is, one can import a very basic packetbeat as is, without any modules (for example to have a flowsbeat connection only flow information).

**Comment by urso** [comment]
Yeah, was a little surprised about this of PR, but I have had to do all modules at once.

For review concentrate on packetbeat/protos/protos.go and registry.go + maybe one module (e.g. http) initialization. Updates to all modules a basically similar:
- create New instead of init
- load config into copy of defaultConfig
- call init (+simplify setFromConfig)

File: packetbeat/main.go (hunk 1 above)
Comment (ruflin) [inline packetbeat/main.go]: By automate I mean the generation of this list based on the directories that exist. In metricbeat I put the list in a different package: https://github.com/elastic/beats/tree/master/metricbeat/include

The advantage of this if someone creates a beat with only one module he can either include all other modules and metricsets he wants one by one or he can just link to `_ "github.com/elastic/beats/metricbeat/include" and gets all the official modules of metricbeat without having to update it all the time.
